File: app/src/main/java/dev/ragnarok/fenrir/model/SwitchableCategory.java
package dev.ragnarok.fenrir.model;

import androidx.annotation.IntDef;

import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;

@IntDef({SwitchableCategory.FRIENDS,
        SwitchableCategory.NEWSFEED_COMMENTS,
        SwitchableCategory.GROUPS,
        SwitchableCategory.PHOTOS,
        SwitchableCategory.VIDEOS,
        SwitchableCategory.MUSIC,
        SwitchableCategory.DOCS,
        SwitchableCategory.BOOKMARKS})
@Retention(RetentionPolicy.SOURCE)
public @interface SwitchableCategory {
    int FRIENDS = 1;
    int NEWSFEED_COMMENTS = 2;
    int GROUPS = 3;
    int PHOTOS = 4;
    int VIDEOS = 5;
    int MUSIC = 6;
    int DOCS = 7;
    int BOOKMARKS = 8;
}

File: app/src/main/kotlin/dev/ragnarok/fenrir/model/SwitchableCategory.kt
package dev.ragnarok.fenrir.model

import androidx.annotation.IntDef

@IntDef(
    SwitchableCategory.FRIENDS,
    SwitchableCategory.NEWSFEED_COMMENTS,
    SwitchableCategory.GROUPS,
    SwitchableCategory.PHOTOS,
    SwitchableCategory.VIDEOS,
    SwitchableCategory.MUSIC,
    SwitchableCategory.DOCS,
    SwitchableCategory.BOOKMARKS
)
@Retention(
    AnnotationRetention.SOURCE
)
annotation class SwitchableCategory {
    companion object {
        const val FRIENDS = 1
        const val NEWSFEED_COMMENTS = 2
        const val GROUPS = 3
        const val PHOTOS = 4
        const val VIDEOS = 5
        const val MUSIC = 6
        const val DOCS = 7
        const val BOOKMARKS = 8
    }
}